### PR TITLE
add upperbound on seed in random

### LINKED ISSUE #6231: Random enters invalid state when large seed is given
To reproduce:
`(Random seed: 16rFFFFFFFFFFFFFFFF) next; inspect`

Now its seed is negative and it outputs funky results.

### PR DESCRIPTION
**Pull request by rakki-18**
Aims to solve #6231

When a large seed is given to Random, it starts giving negative values.
Every seed greater than 1e12 seems to be producing negative values.

This PR adds an upper bound( m = 2^31 - 1) to the value that a seed can take.

### CODE REVIEW

**Review by MarcusDenker — COMMENTED** [review]
I had a look at privateNextSeed and to me this looks like the right thing to be done, yes.

**Comment by MarcusDenker** [comment]
I would prefer to get another review before merging

**Review by MarcusDenker — APPROVED** [review]

**Comment by rakki-18** [comment]
@MarcusDenker Any update on this PR?

**Pull request merged**
